### PR TITLE
Fixes #149, changes the functionality of propagating the error detail…

### LINKED ISSUE #149: [FEATURE] - Mozart Details Field
### Reference Issues

_No response_

### Summary

Mozart provides a details field this should be handled. 

### Unresolved questions

_No response_

### Definition of Done

When details field is propagated to the error response

### PR DESCRIPTION
**Pull request by KristianS93**
## Description

Draft as this changes the functionality a lot. 
Here the response json from Mozart is only serialized to get the result field, the rest is kept as is and served straight to the frontend.

**Needs to be discussed in the group how we want this.**

### Resolved Issue
Fixes #149

## Checklist

Breaks alot of testcases...

### CODE REVIEW

**Comment by KristianS93** [comment]
Converted to draft as this change needs to be discussed within the group.

File: Core/Solutions/Models/SolutionRunnerResponse.cs [hunk 1]
```diff
@@ -4,37 +4,38 @@ namespace Core.Solutions.Models;
...
+// 	string Actual,
+// 	string Expected);
+
+// public record SolutionTestcaseInputParameters(string ValueType, string Value);
+
+// public record MozartResponseDto(
+// 	[property: JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+// 	List<SolutionTestcaseResultDto>? TestCaseResults,
+// 	[property: JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+// 	string? Message);
```
Comment (JMyrtue) [inline Core/Solutions/Models/SolutionRunnerResponse.cs:32]: Suggest deleting this before merge if it is not required any more.

File: IntegrationTest/ExerciseEndpointsTest.cs [hunk 2]
```diff
@@ -323,7 +323,7 @@ public async Task UpdateExercise_FailedToValidateSolution_ShouldReturn_400()
         var solutionRunnerResponse = new SolutionRunnerResponse
         {
             Action = ResponseCode.Failure,
-            ResponseDto = new SolutionResponseDto("failure", null, new List<SolutionTestcaseResultDto> { new SolutionTestcaseResultDto(5, "test", null, null) })
+            // ResponseDto = new SolutionResponseDto("failure", null, new List<SolutionTestcaseResultDto> { new SolutionTestcaseResultDto(5, "test", null, null) })
```
Comment (JMyrtue) [inline IntegrationTest/ExerciseEndpointsTest.cs:326]: Suggest deleting if not needed anymore.

File: IntegrationTest/ExerciseEndpointsTest.cs [hunk 3]
```diff
@@ -349,7 +349,7 @@ public async Task UpdateExercise_CompilationErrorOnSolutionRunner_ShouldReturn_4
         var solutionRunnerResponse = new SolutionRunnerResponse
         {
             Action = ResponseCode.Error,
-            ResponseDto = new SolutionResponseDto("error", "compilation error", null)
+            // ResponseDto = new SolutionResponseDto("error", "compilation error", null)
```
Comment (JMyrtue) [inline IntegrationTest/ExerciseEndpointsTest.cs:352]: Suggest deleting if not needed anymore.

File: IntegrationTest/ExerciseEndpointsTest.cs [hunk 4]
```diff
@@ -474,7 +474,7 @@ public async Task SubmitSolutionProposal_SolutionProposalFailed_ShouldReturn_400
         var solutionRunnerResponse = new SolutionRunnerResponse
         {
             Action = ResponseCode.Failure,
-            ResponseDto = new SolutionResponseDto("failure", null, new List<SolutionTestcaseResultDto> { new SolutionTestcaseResultDto(5, "test", null, null) })
+            // ResponseDto = new SolutionResponseDto("failure", null, new List<SolutionTestcaseResultDto> { new SolutionTestcaseResultDto(5, "test", null, null) })
```
Comment (JMyrtue) [inline IntegrationTest/ExerciseEndpointsTest.cs:477]: Suggest deleting if not needed anymore.

**Review by JMyrtue — APPROVED** [review]
Few comments left regarding removal of outcommented sections.

Approved - left to author of PR to merge.

**Pull request merged**